### PR TITLE
[a11y] Accessibility Enhancements - Group 2

### LINKED ISSUE #3824: [Group 2][Issue] AC-2482::Screen readers not informed when new page view loads. (patte…
This issue is automatically created based on existing pull request: magento/pwa-studio#3803: AC-2482::Screen readers not informed when new page view loads. (patte…

---------
…rn: No announcement for new page view)

## Description

Screen readers not informed when new page view loads.

Environment
Adobe Magento - Venia

Context
Windows 10; Chrome 88; JAWS 2021

Reproduction Steps
Locations (representative sample):

Global Header
Landing Page
Search Results
Product Detail Page
Shopping Bag Mini Cart
Shopping Bag
Checkout - Shipping Information
- Checkout - Confirmation / Create an Account
Sign In
1. Press Tab to move through the content and activate one of the following:

navigation link
Sign In button
Create an Account button
Place Order button
Actual Behavior
When a new page view is rendered, screen reader users are not informed. Although the page title changes, this title change is not announced.

Expected Behavior
Ensure that screen reader users are informed when the page content has changed. This can be accomplished either by providing a live region that is updated when the page title changes (i.e. an element such as a <div> with aria-live="polite") , or by setting focus to the < h1 > of the page.

## Related Issue

Closes https://jira.corp.magento.com/browse/AC-2482

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### LINKED ISSUE #3825: [Issue] AC-2483::Contrast insufficient - custom blue focus indicator
This issue is automatically created based on existing pull request: magento/pwa-studio#3802: AC-2483::Contrast insufficient - custom blue focus indicator

---------

## Description

Contrast insufficient - custom blue focus indicator

Environment
Adobe Magento - Venia

Context
Windows 10; Chrome 88;

Reproduction Steps
Locations (representative sample):

Global Header
Landing Page
Search Results
Product Detail Page
Shopping Bag Mini Cart
Shopping Bag
Checkout - Shipping Information
Checkout - Payment
- Checkout - Confirmation / Create an Account
Sign In
Account - Order History
1. Use a color contrast checker to compare foreground and background colors.

Actual Behavior
The visual focus indicator for buttons, links styled like buttons, or input fields throughout the site is provided by a box-shadow which does not meet the minimum 3:1 contrast ratio for a custom focus indicator. Users with low vision may have trouble identifying this focus indicator. Examples include:

Module 01a - Global Header:
Search input

Module 02 - Landing Page:
Various links styled like buttons, (e.g. "Shop Now" in hero carousel, "Shop Summer" on black background").

Module 03 - Search Results page:
"Filter", "Sort", "See Results" buttons
Checkboxes within filter menu

Module 04 - Product Detail page:
"Add to Cart" button
"Quantity" input field

Module 05a - Shopping Bag dialog/Mini Cart:
"Checkout" button

Module 05b - Shopping Bag:
Buttons, inputs

Module 06a - Checkout - Shipping Information:
Buttons, inputs

Module 06b - Checkout - Payment:
Radio Button

Module 06c - Checkout - Confirmation / Create an Account:
Buttons, inputs

Module 07 - Sign In:
Buttons, inputs

Module 08 - Order History:

Search by order number input & button
Buttons and links styled like buttons:
box-shadow: 6px 6px hsl(var(-brightBlue) / 0.3);
--brightBlue: 233 100% 056%;

The spread and blur of the box shadow combined with the transparency in the hsl color value results in a lower contrast ratio on the page than the assigned hsl value would otherwise provide.

The highest contrast ratio for this style was sampled at #BBC3FF on a #ffffff background, for a contrast ratio of 1.7:1. This contrast ratio falls lower where the focus style is applied on other background colors, such as light gray in the homepage hero carousel (shadow sampled at #BBC3FF on background #f3f3f3 – contrast ratio of ~1.7:1) and dark gray on homepage "Vacations meet fresh style" callout (shadow sampled at #1C2B6C on background #1B252C – contrast ratio of 1.2:1).

Inputs:
box-shadow: 6px 6px rgb(var(-venia-brand-color-1-100));
--venia-global-color-blue-100: 194 200 255;

Against a white background, this box shadow color was sampled at #c2c8ff, for a contrast ratio of 1.6:1.

Expected Behavior
Ensure that custom focus indicators provide a contrast ratio of at least 3:1 against the background color.

Refer to the Color Contrast Checker Tool for assistance: https://www.levelaccess.com/compliance-resource/color-contrast-checker

## Related Issue

Closes https://jira.corp.magento.com/browse/AC-2483

## Verification Steps

✔️ QA Passed
Pre-Conditions:

1. Have Magento instance with sample data installed

2. Make sure to have pwa studio installed

3. Make sure to have a customer login for front end login

 

Manual Steps executed:

Login to venia > Navigate to global header > Search input > "Filter", "Sort", "See Results" buttons
Navigate to Product Detail page >  "Add to Cart" button > "Quantity" input field
Navigate to  shopping Bag dialog/Mini Cart > "Checkout" button
Navigate to Shopping Bag > Buttons, inputs > Checkout - Shipping Information > Buttons, inputs
Navigate to Checkout - Payment > Radio Button > heckout - Confirmation / Create an Account > Buttons, inputs
Navigate to Sign In > Buttons, inputs > Order History > Search by order number input & button > address book > Communication > account information.
 

✖️ Behaviour Before The Fix : The custom focus indicators did not provide a contrast ratio of at least 3:1 against the background color.

image
image

✔️Behaviour After The Fix: The custom focus indicators provide a contrast ratio of at least 3:1 against the background color.

image
image

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### LINKED ISSUE #3826: [Issue] AC-2486::contrast insufficient - medium grey text (Search Results)
This issue is automatically created based on existing pull request: magento/pwa-studio#3801: AC-2486::contrast insufficient - medium grey text (Search Results)

---------

## Description

Contrast insufficient - medium grey text

Environment
Adobe Magento - Venia

Context
Windows 10; Chrome 88;

Reproduction Steps
search "short" - [NODE][body>div:nth-of-type(1)>*:nth-child(1)>*:nth-child(3)]

1. Use a color contrast checker to compare foreground and background colors.

Actual Behavior
There are text elements or images of text that do not meet the WCAG 2.0 required minimum color contrast ratio of 4.5:1 for standard text of 18pt (24px) or 14pt (19px) if bolded. Examples include:

"nn items"
"Items sorted by Best Match"

Foreground:#767B7F
Background:#FFFFFF
The contrast ratio is: 4.3:1

Sufficient contrast ensures that people with low vision or color deficiencies, users viewing the page without color, and users of monochrome screens can see the page content.

Expected Behavior
All text and images of text should provide the following minimum contrast ratios:

For standard text less than 18pt (24px) or less than 14pt (19px) if bolded must have a luminosity contrast ratio of 4.5:1 or more.
For larger text 18pt (24px) or larger or 14pt (19px) if bolded must have a luminosity contrast ratio of 3:1 or more.
Refer to the Color Contrast Checker Tool for assistance:
https://www.levelaccess.com/compliance-resource/color-contrast-checker/

## Related Issue

Closes https://jira.corp.magento.com/browse/AC-2486

## Verification Steps

✔️ QA Passed
Pre-Conditions:

1. Have Magento instance with sample data installed

2. Make sure to have pwa studio installed

3. Make sure to have a customer login for front end login

 

Manual Steps executed:

Login to venia > inspect element and chosde mobile view
navigate to filter and sort button 
observe the  color contrast checker to compare foreground and background colors for the text appearing after sorting or filtering eg "nn items"
"Items sorted by Best Match"
 

✖️ Behaviour Before The Fix : text that do not meet the WCAG 2.0 required minimum color contrast ratio of 4.5:1 for standard text of 18pt (24px) or 14pt (19px) if bolded

image

✔️Behaviour After The Fix: text that do not meet the WCAG 2.0 required minimum color contrast ratio of 4.5:1 for standard text of 18pt (24px) or 14pt (19px) if bolded

image

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### LINKED ISSUE #3828: [Group 2][Issue] BUG#AC-2499::When focusing on 'Venia' image while using screen reader…
This issue is automatically created based on existing pull request: magento/pwa-studio#3799: BUG#AC-2499::When focusing on 'Venia' image while using screen reader…

---------

## Description

When focusing on "Venia" image while using screen reader "Venia Venia" is announced.

Environment
Adobe Magento - Venia

Context
Windows 10; Chrome 88; JAWS 2021

Reproduction Steps
Locations (representative sample):

Global Header
Global Footer
 
1. With JAWS operating, press Tab to move through the content.

Actual Behavior
When focusing on "Venia" image while using screen reader "Venia Venia" is announced. Information provided is too verbose for screen readers.

Module 01a - Global Header
Module 01b - Global Footer

Code Reference:
<a href="/default/">
<div class="image-root-Qpm image-container-_fN">

</div>
</a>

Expected Behavior
Ensure that information is communicated properly to screen reader users. In this case, "Venia" should be announced once.

Consider removing unnecessary title attributes from both images, using null alt text for both images (alt=""), and then adding visually-hidden text within the <a> element to form a concise accessible name for the link. (See sample markup/css below.)

## Related Issue

Closes https://jira.corp.magento.com/browse/AC-2499

## Verification Steps

✔️ QA Passed
 

Pre-Conditions:

1. Have Magento instance with sample data installed

2. Make sure to have pwa studio installed

3. Make sure to have a customer login for front end login

 

Manual Steps executed:

Login to venia > Navigate to global header venia using screen reader > navigate to global footer venia using screen reader

 

✖️ Behaviour Before The Fix : When focusing on "Venia" header anf footer image while using screen reader "Venia Venia" is announced. 

✔️Behaviour After The Fix: Venia" is  announced only once using screen reader.

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### LINKED ISSUE #3830: [Group 2][Issue] AC-2786::Shopping bag button does not programmatically communicate st…
This issue is automatically created based on existing pull request: magento/pwa-studio#3793: AC-2786::Shopping bag button does not programmatically communicate st…

---------

## Description
Shopping bag button does not programmatically communicate state. (Global Header)

**Reproduction Steps**
[NODE][body>div:nth-of-type(1)>*:nth-child(1)>*:nth-child(2)]

1. With JAWS operating and items in your "Bag", press Tab to move through the content.
2. When keyboard focus is on "Bag", press Enter or Space to activate it.

**Actual Behavior**
The Shopping bag button does not provide a programmatic or textual indication of its state. Screen reader users are not informed that the button will expand other content, or whether the associated content is expanded or collapsed.

**Expected Behavior**
Ensure that controls convey their state programmatically or in text. State can be conveyed programmatically with the aria-expanded attribute.

See the WAI-ARIA Authoring Practices design pattern for Disclosure: https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure

## Related Issue
Closes https://jira.corp.magento.com/browse/AC-2786

## Verification Steps
**Pre-Conditions:**

1. Have Magento instance with sample data installed
2. Make sure to have pwa studio installed
3. Make sure to have a customer login for front end login

**Manual Steps executed:**

Login to venia > activate your screen reader
Navigate to global header accessories or other
Add items to your bag > Navigate to shopping bag 
check if the screen reader is properly announcing the behaviour of the bag i.e bag expanded or bag collapsed 

**✖️ Behaviour Before The Fix :** The Shopping bag button does not provide a programmatic or textual indication of its state. Screen reader users are not informed that the button will expand other content, or whether the associated content is expanded or collapsed..

**✔️Behaviour After The Fix:** The Shopping bag button provide a programmatic or textual indication of its state. Screen reader users are  informed that the button will expand other content, or  the associated content is expanded or collapsed..

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### PR DESCRIPTION
**Pull request by tjwiebell**
## Description

Group of PRs merged together for delivery

## Related Issue

* Closes #3824 
* Closes #3825 
* Closes #3826 
* Closes #3828 
* Closes #3830 

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by tjwiebell** [comment]
run cypress

**Review by tjwiebell — COMMENTED** [review]
Merged change set looks good 👍 since I created this merged PR I can't approve, but will send to QA.

I can't reproduce that Cypress failure locally, so it may just be a flaky test.

**Comment by dpatil-magento** [comment]
run cypress

**Comment by dpatil-magento** [comment]
run lighthouse-desktop

**Comment by dpatil-magento** [comment]
run lighthouse-mobile

**Review by dpatil-magento — APPROVED** [review]

**Comment by dpatil-magento** [comment]
QA Approved. Failed cypress not related.

File: packages/venia-ui/lib/components/Footer/__tests__/__snapshots__/footer.spec.js.snap [hunk 1]
```diff
@@ -205,6 +205,7 @@ exports[`footer renders copyright 1`] = `
       foo
     </p>
     <a
+      aria-label="Venia"
```
Comment (annick-primard) [inline packages/venia-ui/lib/components/Footer/__tests__/__snapshots__/footer.spec.js.snap:208]: This should also mention the destination of the link: return to homepage.

**Pull request merged**